<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Modbus_Window</class>
 <widget class="QMainWindow" name="Modbus_Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>999</width>
    <height>850</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>221</width>
      <height>220</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="title">
     <string>串口配置</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>50</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>串口号：</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxParity">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>140</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
     <item>
      <property name="text">
       <string>无校验</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>奇校验</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>偶校验</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBoxStop">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>180</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <item>
      <property name="text">
       <string>1位</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1.5位</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2位</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBoxBaud">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>8</number>
     </property>
     <item>
      <property name="text">
       <string>4800</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>9600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>14400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>19200</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>28800</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>38400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>56000</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>57600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>115200</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>128000</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>230400</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>256000</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>50</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>校验位：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>50</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>停止位：</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxData">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>100</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <item>
      <property name="text">
       <string>5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>9</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>50</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>数据位：</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxPort">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>50</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>波特率：</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>239</x>
      <y>10</y>
      <width>751</width>
      <height>400</height>
     </rect>
    </property>
    <property name="title">
     <string>动态曲线</string>
    </property>
    <widget class="QChartView" name="graphicsView">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>770</width>
       <height>371</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>239</x>
      <y>420</y>
      <width>751</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>收发显示</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>770</width>
       <height>200</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>221</width>
      <height>481</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="title">
     <string>交互配置</string>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>22</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>从站地址：</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxDataType">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>141</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <item>
      <property name="text">
       <string>0 Coils</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1 Input Status</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 Input Registers</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3 Holding Registers</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>读取数量：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>起始地址：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>数据类型：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxStation">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>101</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>254</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxStartRead">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>140</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNumberRead">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>180</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>125</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxAuto">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>自动读取</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxInterval">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>220</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>3600000</number>
     </property>
     <property name="value">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>间隔周期：</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonRead">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>280</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>  读取</string>
     </property>
     <property name="icon">
      <iconset resource="Img.qrc">
       <normaloff>:/Img/images/books.png</normaloff>:/Img/images/books.png</iconset>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxStartWrite">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>320</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>320</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>起始地址：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>360</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>写入数量：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNumberWrite">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>360</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>125</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonWrite">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>440</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>  写入</string>
     </property>
     <property name="icon">
      <iconset resource="Img.qrc">
       <normaloff>:/Img/images/Write.png</normaloff>:/Img/images/Write.png</iconset>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEditWriteValue">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>400</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>400</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>写入数值：</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBoxFormat">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>100</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>报文格式：</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="MySQLcheckBox">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>250</y>
       <width>110</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>保存到数据库</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="HEXCheckBox">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>280</y>
       <width>91</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>输出16进制</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>239</x>
      <y>660</y>
      <width>751</width>
      <height>60</height>
     </rect>
    </property>
    <property name="title">
     <string>操作区域</string>
    </property>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>54</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>发送命令：</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEditSend">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>561</width>
       <height>30</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonSend">
     <property name="geometry">
      <rect>
       <x>640</x>
       <y>20</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>  发送</string>
     </property>
     <property name="icon">
      <iconset resource="Img.qrc">
       <normaloff>:/Img/images/send.png</normaloff>:/Img/images/send.png</iconset>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="BackBtn">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>720</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font-family:'仿宋';
font-size:20px;
color: #2f3640;
background-color: #eea2a4;
border-color: #eea2a4;
border-radius: 20px;
border-style: solid;
border-width: 2px;
padding: 5px;</string>
    </property>
    <property name="text">
     <string>返回</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>999</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>设备</string>
    </property>
    <addaction name="actionConnect"/>
    <addaction name="actionDisconnect"/>
    <addaction name="actionRefresh"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>工具</string>
    </property>
    <addaction name="actionConfig"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionConnect"/>
   <addaction name="actionDisconnect"/>
   <addaction name="actionRefresh"/>
   <addaction name="separator"/>
   <addaction name="actionSave"/>
   <addaction name="actionConfig"/>
  </widget>
  <action name="actionConnect">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/connect.png</normaloff>:/Img/images/connect.png</iconset>
   </property>
   <property name="text">
    <string>连接</string>
   </property>
   <property name="toolTip">
    <string>连接从站</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/disconnect.png</normaloff>:/Img/images/disconnect.png</iconset>
   </property>
   <property name="text">
    <string>断开</string>
   </property>
   <property name="toolTip">
    <string>断开连接</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/exit.png</normaloff>:/Img/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出软件</string>
   </property>
  </action>
  <action name="actionConfig">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/settings.png</normaloff>:/Img/images/settings.png</iconset>
   </property>
   <property name="text">
    <string>配置</string>
   </property>
   <property name="toolTip">
    <string>配置参数</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/save.png</normaloff>:/Img/images/save.png</iconset>
   </property>
   <property name="text">
    <string>存档</string>
   </property>
   <property name="toolTip">
    <string>报文保存</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="icon">
    <iconset resource="Img.qrc">
     <normaloff>:/Img/images/refresh.png</normaloff>:/Img/images/refresh.png</iconset>
   </property>
   <property name="text">
    <string>刷新</string>
   </property>
   <property name="toolTip">
    <string>刷新串口列表</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Img.qrc"/>
 </resources>
 <connections/>
</ui>
